server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -561,6 +562,15 @@ protected CompactionMetadata reserveCompaction(CompactionJobQueues.MetaJob metaJ
         var tabletMutator = tabletsMutator.mutateTablet(extent).requireAbsentOperation()
             .requireSame(tabletMetadata, FILES, SELECTED, ECOMP);
 
+        if (metaJob.getJob().getKind() == CompactionKind.SYSTEM) {
+          var selectedFiles = tabletMetadata.getSelectedFiles();
+          if (selectedFiles != null && (selectedFiles.getCompletedJobs() == 0
+              && !Collections.disjoint(jobFiles, selectedFiles.getFiles()))) {
```

Comment:
May be able to use the ` getFilesResevedBySelection` function mentioned in another comment to simplify this a bit and consider expiration.

Reply:
Added in https://github.com/apache/accumulo/commit/dafdc87afcab2dee407f8d2fc4953f148963feb1